server/src/main/java/org/opensearch/action/main/MainResponse.java

```diff
@@ -102,7 +102,11 @@ public Build getBuild() {
     @Override
     public void writeTo(StreamOutput out) throws IOException {
         out.writeString(nodeName);
-        Version.writeVersion(version, out);
+        if (out.getVersion().before(Version.V_1_0_0)) {
```

Comment:
May be a dumb question, are we saying any version before 1.0.0 will write 7.10.2?
Wouldn't this break for older clients

Reply:
Any client version before V_1_0_0, yes; which will pass with older clients because they will report a version ID less than OpenSearch 1.0.0 (e.g., ES 7.10.2 version id is 7100299)